frontend/src/pages/checklist/[filters].tsx

```diff
@@ -125,6 +126,28 @@ const ChecklistResults: FC<ChecklistResultsProps> = ({
     })
   }
 
+  function handleExpandAllToggle() {
+    if (expandAll) {
+      setExpandedGroups([])
+      setExpandedTasks([])
+    } else {
+      setExpandedGroups([beforeRetiring.id, applyingBenefits.id, receivingBenefits.id])
+      setExpandedTasks(
+        [...beforeRetiring.tasks, ...applyingBenefits.tasks, ...receivingBenefits.tasks].map((t) => t.id)
+      )
+    }
+    setExpandAll((prev) => !prev)
+    const encodedFilters = encodeURIComponent(window.btoa(JSON.stringify(filters)))
+    router.replace(
+      { pathname: `/checklist/${encodedFilters}`, query: { group: expandedGroups, task: expandedTasks } },
```

Comment:
I will adjust the other places to match this pattern in another PR.

```suggestion
      { pathname: router.pathname, query: { ...router.query, group: expandedGroups, task: expandedTasks } },
```